MantidQt/MantidWidgets/src/DataProcessorUI/GenericDataProcessorPresenter.cpp

```diff
@@ -520,53 +521,104 @@ std::string GenericDataProcessorPresenter::getPostprocessedWorkspaceName(
   return prefix + boost::join(outputNames, "_");
 }
 
-/**
-Loads a run from disk or fetches it from the AnalysisDataService
-@param run : The name of the run
-@param instrument : The instrument the run belongs to
-@param prefix : The prefix to be prepended to the run number
-@throws std::runtime_error if the run could not be loaded
-@returns a shared pointer to the workspace
-*/
+/** Loads a run found from disk or AnalysisDataService
+ *
+ * @param run : The name of the run
+ * @param instrument : The instrument the run belongs to
+ * @param prefix : The prefix to be prepended to the run number
+ * @throws std::runtime_error if the run could not be loaded
+ * @returns a shared pointer to the workspace
+ */
 Workspace_sptr
-GenericDataProcessorPresenter::loadRun(const std::string &run,
-                                       const std::string &instrument,
-                                       const std::string &prefix) {
+GenericDataProcessorPresenter::getRun(const std::string &run,
+                                      const std::string &instrument,
+                                      const std::string &prefix) {
+
+  bool runFound;
+  std::string outName;
+  std::string fileName = instrument + run;
+
+  outName = findRunInADS(run, prefix, runFound);
+  if (!runFound ||
+      AnalysisDataService::Instance().retrieveWS<IEventWorkspace>(outName) !=
+          NULL) {
+    // Workspace must not be an event workspace
```

Comment:
I'd suggest that, if possible, we should keep all functionality related to event workspaces outside the `GenericDataProcessorPresenter`. The reason is that this class is only meant to deal with non-event runs, that is why we added event functionality to a separate class `ReflGenericDataProcessorPresenter`. I think we should try to avoid mixing event and non-event workspaces here. Is there any reason why this is not possible?